### ISSUE TITLE
F1 score in devise_hype.py training code

### ISSUE THREAD

**Issue by VenkteshV** [ShaoTengLiu/Hyperbolic_ZSL#2]
HIi, thanks for releasing the code. I had one query. wouldn't the code calculating f1 score in train and val loop throw an error as it is being computed between input and label representations which is multioutput continuous and is not supported by sklearn ?

**Comment by 2653888596** [comment]
HI,VenkteshV, I had one question ,could you please release your data? I don't know what format of data, if you can release your data processing , I vert grateful, My E-mail is 2653888596@qq.com. Thanks for you!

**Comment by 2653888596** [comment]
> HIi, thanks for releasing the code. I had one query. wouldn't the code calculating f1 score in train and val loop throw an error as it is being computed between input and label representations which is multioutput continuous and is not supported by sklearn ?

HI,VenkteshV, I had one question ,could you please release your data? I don't know what format of data, if you can release your data processing , I vert grateful, My E-mail is 2653888596@qq.com. Thanks for you!

**Comment by VenkteshV** [comment]
> > HIi, thanks for releasing the code. I had one query. wouldn't the code calculating f1 score in train and val loop throw an error as it is being computed between input and label representations which is multioutput continuous and is not supported by sklearn ?
> 
> HI,VenkteshV, I had one question ,could you please release your data? I don't know what format of data, if you can release your data processing , I vert grateful, My E-mail is [2653888596@qq.com](mailto:2653888596@qq.com). Thanks for you!

Hi . I cannot share the data right now. However will do that once the work is ready. Thanks for expressing interest.

**Comment by CallMeNumber1** [comment]
> > > HIi, thanks for releasing the code. I had one query. wouldn't the code calculating f1 score in train and val loop throw an error as it is being computed between input and label representations which is multioutput continuous and is not supported by sklearn ?
> > 
> > 
> > HI,VenkteshV, I had one question ,could you please release your data? I don't know what format of data, if you can release your data processing , I vert grateful, My E-mail is [2653888596@qq.com](mailto:2653888596@qq.com). Thanks for you!
> 
> Hi . I cannot share the data right now. However will do that once the work is ready. Thanks for expressing interest.

I also encountered the question as you did. I am wondering how you solved the error when calculating f1 score. Looking forward to your relpy.

**Comment by MinaGhadimiAtigh** [comment]
I checked the code and encountered the same error. 

I believe the f1 score is not used properly in the code because the input is a continuous label representation and not a meaningful input to calculate f1. 

A better solution can be performing classification and the calculate f1 score given correct labels and predicted labels.
